assets/js/wp-user-manager.js

```diff
@@ -165,4 +167,116 @@ jQuery( function( $ ) {
+			action: 'wpum_delete_profile_image',
+		},
+		function( data, status ){
+
+		});
+	}
+
+	// load FilePond resources
+	var resources = [
+		'plugin/filepond-encode.min.js',
+		'plugin/filepond-preview.css',
+		'plugin/filepond-preview.min.js',
+		'plugin/filepond-crop.min.js',
+		'plugin/filepond-resize.min.js',
+		'filepond.min.css',
+		'filepond.min.js',
+	].map(function(resource) { return wpumFrontend.pluginurl + 'assets/js/vendor/filepond/' + resource });
+
+	loadResources(resources).then(function() {
+
+		// register plugins
+		FilePond.registerPlugin(
+			FilePondPluginFileEncode,
+			FilePondPluginImagePreview,
+			FilePondPluginImageCrop,
+			FilePondPluginImageResize,
+		);
+
+		const defaultCover = $('.fieldset-user_cover .wpum-uploaded-files input');
+		const FilePondCover = FilePond.create( $('.fieldset-user_cover input[type="file"]')[0], {
+			stylePanelLayout: 'compact square',
+			server: {
+				url: wpumFrontend.ajaxurl,
+				process: {
+					method: 'POST',
+					ondata: (formData) => {
+						formData.append( 'action', 'wpum_upload_profile_image' );
+						formData.append( 'key', 'user_cover');
+						formData.append( 'nonce', $( '#account_update_nonce' ).val() );
+						return formData;
+					}
+				},
+				load: '?action=wpum_load_profile_image&type=user_cover_path&',
+			}
+		});
+
+		if ( defaultCover.length > 0 ) {
+			FilePondCover.files = [ {
+				source: defaultCover.val(),
+				options: {
+					type: 'local',
+					metadata: {
+						poster: defaultCover.length > 0 ? defaultCover.val() : null
+					}
+				}
+			} ];
+		}
+
+		const defaultAvatar = $('.fieldset-user_avatar .wpum-uploaded-files input');
+		const FilePondAvatar = FilePond.create( $('.fieldset-user_avatar input[type="file"]')[0], {
+			imageResizeTargetWidth: 250,
+			imageResizeTargetHeight: 250,
+			styleLoadIndicatorPosition: 'center bottom',
+			styleProgressIndicatorPosition: 'center bottom',
+			styleButtonRemoveItemPosition: 'center bottom',
+			styleButtonProcessItemPosition: 'center bottom',
+			stylePanelLayout: 'compact circle',
+			server: {
+				url: wpumFrontend.ajaxurl,
+				process: {
+					method: 'POST',
+					ondata: (formData) => {
+						formData.append( 'action', 'wpum_upload_profile_image' );
+						formData.append( 'key', 'user_avatar' );
+						formData.append( 'nonce', $( '#account_update_nonce' ).val() );
+						return formData;
+					}
+				},
+				load: defaultAvatar.length > 0 ? '?action=wpum_load_profile_image&type=current_user_avatar_path&' : null,
+			}
+		});
+
+		if ( defaultAvatar.length > 0 ) {
+			FilePondAvatar.files = [ {
+				source: defaultAvatar.val(),
+				options: {
+					type: 'local',
+					metadata: {
+						poster: defaultCover.length > 0 ? defaultCover.val() : null
+					}
+				}
+			} ];
+		}
+
+		const filePondCover = document.querySelector('.fieldset-user_cover .filepond--root');
+		filePondCover.addEventListener('FilePond:removefile', e => {
+			$('.fieldset-user_cover .wpum-uploaded-files').html('');
```

Comment:
Can we add support for any image file field (https://github.com/WPUserManager/wpum-custom-fields/issues/42)

Reply:
@polevaultweb 
This is done.